keps/sig-node/3085-pod-conditions-for-starting-completition-of-sandbox-creation/README.md

```diff
@@ -0,0 +1,1441 @@
+unavailable (before the sandbox creation condition can be patched) and Kubelet
+crashes or restarts (shortly after API server becoming and staying unavailable),
+the `lastTransitionTime` field may be inaccurate. This is described in the
+section [above](#unavailability-of-api-server-or-etcd-along-with-kubelet-restart).
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+None so far
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+SLOs are not applicable to pod status fields. Overall Kubernetes node level SLOs
+may leverage this feature.
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+The main drawback associated with the new pod sandbox conditions involves a
+slight potential increase in calls to the API Server from Kubelet to patch
+`status` = `true` for the new `SandboxReady` condition in a pod's status.
+Typically, this would involve an extra patch call for pod status in the lifetime
+of most pods (if the status manager does not batch them with other pod status
+updates): one when pod sandbox creation completes and another when the pod is
+terminated. However, there could be a higher number of patch calls to API Server
+if the pod sandbox environment (like a microvm) starts successfully and then
+crashes in a re-start loop.
+
+Caching of updates to pod status by the pod status manager and batching pod
+status updates (which is already in place) can help mitigate frequent patch
+calls to API server.
+
+## Alternatives
+
+<!--
+What other approaches did you consider, and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
```

Comment:
Two other options: use an [extended condition](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate) rather than `SandboxReady`.
eg `kubernetes.io/pod-sandbox-ready`

_or_

report the individual sandbox elements and their readiness
eg `kubernetes.io/pod-network-ready`, `kubernetes.io/pod-kernel-namespace-ready` etc

Reply:
Added ^ as an alternative.